### PR TITLE
A unit Test to make sure ClockedSchedule and PeriodicTasks are shown in TimeZone format

### PR DESCRIPTION
**Pull request by oreaba**
This is a unit test for merged pull request #464

### CODE REVIEW

**Comment by oreaba** [comment]
@auvipy Here is the unit test

File: t/unit/test_models.py [hunk 1]
```diff
@@ -146,3 +148,12 @@ class ClockedScheduleTestCase(TestCase, TestDuplicatesMixin):
...
         self._test_duplicate_schedules(ClockedSchedule, kwargs)
+
+    # we must have a valid the time-zone (not UTC) to do a correct test 
+    @override_settings(TIME_ZONE='Africa/Cairo') 
+    def test_timezone_format(self):
+        """Make sure the scheduled time is not shown in UTC when time zone is used"""
+        tz_info             = pytz.timezone(settings.TIME_ZONE).localize(datetime.datetime.utcnow())
+        schedule, created   = ClockedSchedule.objects.get_or_create(clocked_time=tz_info)
+        self.assertEquals(str(schedule.clocked_time), str(schedule))    # testnig str(schedule) calls make_aware() internally
+
```
Comment (auvipy) [inline t/unit/test_models.py:159]: no new line

File: t/unit/test_models.py [hunk 2]
```diff
@@ -146,3 +148,12 @@ class ClockedScheduleTestCase(TestCase, TestDuplicatesMixin):
...
         kwargs = {'clocked_time': timezone.now()}
         self._test_duplicate_schedules(ClockedSchedule, kwargs)
+
+    # we must have a valid the time-zone (not UTC) to do a correct test 
+    @override_settings(TIME_ZONE='Africa/Cairo') 
+    def test_timezone_format(self):
+        """Make sure the scheduled time is not shown in UTC when time zone is used"""
+        tz_info             = pytz.timezone(settings.TIME_ZONE).localize(datetime.datetime.utcnow())
+        schedule, created   = ClockedSchedule.objects.get_or_create(clocked_time=tz_info)
+        self.assertEquals(str(schedule.clocked_time), str(schedule))    # testnig str(schedule) calls make_aware() internally
```
Comment (auvipy) [inline t/unit/test_models.py:158]: also assert should be in pytest style for any celery project

**Review by auvipy — CHANGES_REQUESTED** [review]

File: t/unit/test_models.py (hunk 2 above)
Comment (oreaba) [reply t/unit/test_models.py:158]: @auvipy 
when I run `pytest ` I get 97 errors, although when I run `pytest --fixtures ` it says collected 97 items.
any cluses?

Comment (oreaba) [reply t/unit/test_models.py:158]: I get "fixture 'patching' not found"

Comment (oreaba) [reply t/unit/test_models.py]: I have updated the assert and pushed a commit.
Please let me know if you need anything. 
Also, please let me know how to pytest it without errors.

File: t/unit/test_models.py [hunk 3]
```diff
@@ -146,3 +148,12 @@ class ClockedScheduleTestCase(TestCase, TestDuplicatesMixin):
...
         self._test_duplicate_schedules(ClockedSchedule, kwargs)
+
+    # IMPORTANT: we must have a valid the time-zone (not UTC) to do an accurate testing
+    @override_settings(TIME_ZONE='Africa/Cairo') 
+    def test_timezone_format(self):
+        """Make sure the scheduled time is not shown in UTC when time zone is used"""
+        tz_info             = pytz.timezone(settings.TIME_ZONE).localize(datetime.datetime.utcnow())
+        schedule, created   = ClockedSchedule.objects.get_or_create(clocked_time=tz_info)
+        # testnig str(schedule) calls make_aware() internally
+        assert str(schedule.clocked_time) == str(schedule)
```
Comment (auvipy) [inline t/unit/test_models.py:159]: please remove extra spaces

**Review by auvipy — CHANGES_REQUESTED** [review]
use any linter for the style issues

File: t/unit/test_models.py [hunk 4]
```diff
@@ -146,3 +148,12 @@ class ClockedScheduleTestCase(TestCase, TestDuplicatesMixin):
     def test_duplicate_schedules(self):
         kwargs = {'clocked_time': timezone.now()}
         self._test_duplicate_schedules(ClockedSchedule, kwargs)
+
+    # IMPORTANT: we must have a valid the time-zone (not UTC) to do an accurate testing
+    @override_settings(TIME_ZONE='Africa/Cairo') 
+    def test_timezone_format(self):
+        """Make sure the scheduled time is not shown in UTC when time zone is used"""
+        tz_info             = pytz.timezone(settings.TIME_ZONE).localize(datetime.datetime.utcnow())
```
[the remark this answers is not in the document]
Comment (oreaba) [reply t/unit/test_models.py]: @auvipy 
Formatted correctly now
but I still can not call pytest to perform the testing.

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+    "python.testing.unittestEnabled": false,
+    "python.testing.pytestEnabled": true
+}
```
Comment (auvipy) [inline .vscode/settings.json:4]: this file should be in your gitignore and not committed

File: t/unit/test_models.py [hunk 6]
```diff
@@ -153,7 +153,7 @@ def test_duplicate_schedules(self):
...
     def test_timezone_format(self):
         """Make sure the scheduled time is not shown in UTC when time zone is used"""
-        tz_info             = pytz.timezone(settings.TIME_ZONE).localize(datetime.datetime.utcnow())
-        schedule, created   = ClockedSchedule.objects.get_or_create(clocked_time=tz_info)
+        tz_info = pytz.timezone(settings.TIME_ZONE).localize(datetime.datetime.utcnow())
+        schedule, created = ClockedSchedule.objects.get_or_create(clocked_time=tz_info)
         # testnig str(schedule) calls make_aware() internally
-        assert str(schedule.clocked_time) == str(schedule)        
\ No newline at end of file
+        assert str(schedule.clocked_time) == str(schedule)
```
Comment (auvipy) [inline t/unit/test_models.py:159]: please get rid of no newline

**Review by auvipy — CHANGES_REQUESTED** [review]

File: t/unit/test_models.py (hunk 3 above)
Comment (oreaba) [reply t/unit/test_models.py]: @auvipy done

**Comment by auvipy** [comment]
were you able to run test locally?

**Comment by oreaba** [comment]
> @auvipy Yes I ran the test successfully, but this was only using 'python manage.py test'
pytest did not work with me although I have installed the requirements.
It fails at 
 \django-celery-beat\t\unit\conftest.py, line 29

fixture 'patching' not found

if you can help please let me know.
Thanks

**Comment by auvipy** [comment]
please wait untill i make the CI work. i might need to push to your branch as well

**Comment by oreaba** [comment]
> please wait untill i make the CI work. i might need to push to your branch as well

Hi @auvipy 
Any updates? should we merge this unit test?

**Comment by auvipy** [comment]
I have started CI to test this properly

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
> v 2.3.0 of beat segfaults in our cloudfoundry env, with an error that suggests that this unit test is failing.
> 
> `11:07:42.847: [APP/PROC/WEB.0] ERROR 2022-06-10 15:07:42,845 none 53 --- 280 [celery.utils.dispatch.signal] send : Signal handler <bound method DjangoFixup.on_import_modules of <celery.fixups.django.DjangoFixup object at 0x7f60c1d25ac0>> raised: ValueError("Extraneous characters at end of TZ string: b'<+12>-12<+13>,M11.1.0,M1.2.2/123'")`

can you work on a fix?

**Pull request merged**